### DISCUSSION TITLE
global filter lazy load not working

### DISCUSSION THREAD

**Discussion by johnlauwers74** [primefaces/community#1341] [PrimeVue]
I have a datatable filled. It is connected to a backend server.
Field sorting and field filtering wortks perfectly.
But the global search does not work.
The onFilter event does not fire when I type something in the global filter inputtext.

`<DataTable
            :value="soortboekingen" lazy paginator :first="first" :rows="10" v-model:filters="filters" ref="dt" dataKey="id"
            :totalRecords="totalRecords" :loading="loading" @page="onPage($event)" @sort="onSort($event)" @filter="onFilter($event)" filterDisplay="row"
            :globalFilterFields="['soort','kasboek']"
             tableStyle="min-width: 75rem"
        >
          <template #header>
            <div class="flex justify-content-between">
              <div class="my-2">
                <Button label="Nieuw" icon="pi pi-plus" class="mr-2" severity="success" @click="store.setShow('add', null)" />
              </div>
              <div class="my-2">
                <span class="p-input-icon-left w-full mb-4">
                    <i class="pi pi-search"></i>
                    <InputText v-model="filters['global'].value"  placeholder="Keyword Search" />

                </span>
              </div>

            </div>
          </template>
          <Column field="soort" header="Soort" filterMatchMode="startsWith" :sortable="true" headerStyle="width:33%; min-width:10rem;">
            <template #filter="{filterModel,filterCallback}">
              <InputText type="text" v-model="filterModel.value" @keydown.enter="filterCallback()" class="p-column-filter" placeholder="Search"/>
            </template>
            <template #body="slotProps">
              <span class="p-column-title">Code</span>
              {{ slotProps.data.soort }}
            </template>
          </Column>
          <Column field="kasboek" header="Kasboek" filterMatchMode="equals" dataType="boolean" headerStyle="width:33%; min-width:10rem;">
            <template #filter="{ filterModel, filterCallback }">
              <TriStateCheckbox v-model="filterModel.value" @change="filterCallback()" />
            </template>
            <template #body="slotProps">
              <i class="pi" style="font-size: 1.5rem" :class="{ 'pi-check-circle text-green-500': slotProps.data.kasboek, 'pi-times-circle text-red-400': !slotProps.data.kasboek }"></i>
            </template>

          </Column>
          <Column headerStyle="min-width:10rem;">
            <template #body="slotProps">
              <Button icon="pi pi-pencil" class="mr-2" severity="success" rounded @click="store.setShow('edit',slotProps.data)" />
              <Button icon="pi pi-trash" class="mt-2" severity="warning" rounded @click="store.setShowDelete(slotProps.data)" />
            </template>
          </Column>
        </DataTable>`
`const filters = ref({
  'global': { value: null, matchMode: FilterMatchMode.CONTAINS },
  'soort': {value: '', matchMode: FilterMatchMode.CONTAINS},
  'kasboek': {value: null, matchMode: FilterMatchMode.EQUALS}
});`

Can anybody help what I am doing wrong?

**Comment by tugcekucukoglu** [comment]
Could you share a reproducer link?
